quickstarts/azure-openai-assistants/commonjs/ts/src/assistant.ts

```diff
@@ -0,0 +1,67 @@
+import { AzureOpenAI } from 'openai';
+import {
+  Assistant,
+  AssistantCreateParams,
+  AssistantTool
+} from 'openai/resources/beta/assistants';
+import { Message, MessagesPage } from 'openai/resources/beta/threads/messages';
+import { Run } from 'openai/resources/beta/threads/runs/runs';
+import { Thread } from 'openai/resources/beta/threads/threads';
+
+export async function assistant(
+  assistantsClient: AzureOpenAI,
+  azureOpenAIDeployment: string
+): Promise<void> {
+  const options: AssistantCreateParams = {
+    model: azureOpenAIDeployment, // Deployment name seen in Azure AI Studio
+    name: 'Math Tutor',
+    instructions:
+      'You are a personal math tutor. Write and run JavaScript code to answer math questions.',
+    tools: [{ type: 'code_interpreter' } as AssistantTool]
+  };
+  const role = 'user';
+  const message =
+    'I need to solve the equation `3x + 11 = 14`. Can you help me?';
+
+  // Create an assistant
+  const assistantResponse: Assistant =
+    await assistantsClient.beta.assistants.create(options);
+  console.log(`Assistant created: ${JSON.stringify(assistantResponse)}`);
+
+  // Create a thread
+  const assistantThread: Thread = await assistantsClient.beta.threads.create(
+    {}
+  );
+  console.log(`Thread created: ${JSON.stringify(assistantThread)}`);
+
+  // Add a user question to the thread
+  const threadResponse: Message =
+    await assistantsClient.beta.threads.messages.create(assistantThread.id, {
+      role,
+      content: message
+    });
+  console.log(`Message created:  ${JSON.stringify(threadResponse)}`);
+
+  // Run the thread and poll it until it is in a terminal state
+  const runResponse: Run =
+    await assistantsClient.beta.threads.runs.createAndPoll(
+      assistantThread.id,
+      {
+        assistant_id: assistantResponse.id
+      },
+      { pollIntervalMs: 500 }
+    );
+  console.log(`Run created:  ${JSON.stringify(runResponse)}`);
+
+  // Get the messages
+  const runMessages: MessagesPage =
+    await assistantsClient.beta.threads.messages.list(assistantThread.id);
+  for await (const runMessageDatum of runMessages) {
+    for (const item of runMessageDatum.content) {
+      // types are: "image_file" or "text"
+      if (item.type === 'text') {
```

Comment:
should we handle all types? there is also `image_url`.

Reply:
Can you tell me more about all types in terms of why someone would do this? I'm not that familiar with this use case so I'm missing some of the details to understand your comment.

Reply:
You can include images in messages, see https://platform.openai.com/docs/api-reference/messages/createMessage. You can ask the assistant to describe the images for example. Here is an overview of the vision scenario: https://platform.openai.com/docs/guides/vision